packages/web-components/fast-element/src/templating/repeat.ts

```diff
@@ -220,8 +223,8 @@ export class RepeatBehavior<TSource = any> implements ViewBehavior, Subscriber {
         if (this.directive.options.positioning) {
             for (let i = 0, ii = views.length; i < ii; ++i) {
                 const context = views[i].context;
-                context.length = i;
-                context.index = ii;
+                context.length = ii;
+                context.index = i;
```

Comment:
This is the actual fix. Everything else is test improvements to catch this.

Reply:
This sort of double assignment in a for loop is very unconventional. Is this approach used for performance reasons? Could it be rewritten to improve readability?

Reply:
would renaming `ii` to something like `viewsLength` make sense? That would go a long way for improving readability without affecting the routine.

Reply:
Yes. I was just thinking the same thing. Because the bug is so hard to see with the current variable names, which is probably why it happened in the first place. I'll make that improvement in just a sec.

Reply:
Pushed the change. Thanks @radium-v for the feedback!